### PR TITLE
Fix tests

### PR DESCRIPTION
**Pull request by WinterSilence**
Add assets repository - you can omit repository in `composer.json` of extension packages
Use filesystem methods

| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌
| Fixed issues  |

### CODE REVIEW

**Comment by samdark** [comment]
> you can omit repository in composer.json of extension packages

That's very implicit. I'd avoid that.

**Comment by rob006** [comment]
Also, it won't work for fresh projects. In order to make it work, you need `yii2-composer` plugin installed. But in order to install it, you need to resolve dependencies. And you need `https://asset-packagist.org` for that. So you have chicken-egg problem.

That is the main reason why fxp asset plugin needs to be installed globally.

**Comment by WinterSilence** [comment]
@rob006 do you try this?

**Comment by rob006** [comment]
@WinterSilence No, I did not tested this, but I'm not sure how it could work for Yii 2 projects. Did you tested it in fresh project without `composer.lock`?

**Comment by WinterSilence** [comment]
@rob006 I don't see your test, you can omit repository for extensions, in most cases it's happened only you create new extension

**Comment by samdark** [comment]
@rob006 fxp is thing of the past, we've switched to asset packaist long ago.

File: Plugin.php [hunk 1]
```diff
@@ -45,14 +45,22 @@ class Plugin implements PluginInterface, EventSubscriberInterface
      */
     public function activate(Composer $composer, IOInterface $io)
     {
+        // Global assets repository
+        $repositoryManager = $composer->getRepositoryManager();
+        $repositoryManager->addRepository($repositoryManager->createRepository(
+            'composer',
+            ['url' => 'https://asset-packagist.org'],
+            'asset-packagist.org'
+        ));
```
Comment (samdark) [inline Plugin.php:54]: I don't think this part should be done.

Comment (WinterSilence) [reply Plugin.php:54]: @samdark why?

Comment (samdark) [reply Plugin.php:54]: 1) Implicit.
2) It fixes a problem that doesn't exist. Extensions have it already.

Comment (WinterSilence) [reply Plugin.php:54]: @samdark 1. why need define repository in extensions that they don't use? 2. It's already have only in existed extensions of current project

Comment (samdark) [reply Plugin.php:54]: @WinterSilence the fact that you have a repository added to a package doesn't automatically add this repository to your main application. This change in the plugin does exactly that — always adding the repository no matter if it is used or not. I may choose to use fxp (yes, it's our past but still there are projects still using it directly) instead and then adding repository will not be desired.

Comment (WinterSilence) [reply Plugin.php:54]: @samdark you still can use it https://getcomposer.org/doc/articles/repository-priorities.md#repository-priorities

Comment (samdark) [reply Plugin.php]: But you can not omit it anymore.

Comment (WinterSilence) [reply Plugin.php]: @samdark you can't omit it's any way

File: Plugin.php [hunk 2]
```diff
@@ -45,14 +45,22 @@ class Plugin implements PluginInterface, EventSubscriberInterface
...
     public function activate(Composer $composer, IOInterface $io)
     {
+        // Add assets repository globally
+        $repositoryManager = $composer->getRepositoryManager();
+        $repositoryManager->addRepository($repositoryManager->createRepository(
+            'composer',
+            ['url' => 'https://asset-packagist.org'],
+            'assets'
+        ));
+
```
Comment (samdark) [inline Plugin.php:55]: Doesn't seem to belong to this PR.

```suggestion
```
